storage/spider/scripts/uninstall_spider.sql

```diff
@@ -0,0 +1,20 @@
+# Copyright (C) 2020 Otto Kekäläinen
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1335  USA
+
+drop procedure if exists mysql.spider_plugin_installer;
+DELETE FROM INFORMATION_SCHEMA.plugins WHERE PLUGIN_NAME = 'SPIDER';
```

Comment:
information schema deletes? Are you sure? this should populate from installed plugins. I've never seen a delete on an information_schema do anything and I'm pretty sure its not plugged in to do anything apart from SELECT.

I would of thought a `UNINSTALL` would of removed the plugin from mysql.plugin.

Is `FLUSH TABLES` for the spider tables? Should it be before the uninstall? Does uninstall fail if spider tables are open? Global for all tables seems excessive (could leave this as a TODO to be specific to spider tables based on WITH READ LOCK comment in kb). If used include `FLUSH NO_WRITE_TO_BINLOG TABLE {table list} WITH READ LOCK` https://mariadb.com/kb/en/flush/#the-purpose-of-flush-tables-table_list-with-read-lock if you need the tables closed and to stay closed while the plugin is removed. Flush release occurs at the end of the session.

Reply:
I copied this from what the Mroonga seems to be doing. So maybe a UNINSTALL alone would be enough? Ideally we would not need any script at all, but do just line Connect and OQgraph do with a single config file, but I did not figure out where in CMake those files are generated (and no tips from Anel yet in https://jira.mariadb.org/browse/MDEV-19917).

Reply:
I agree, better not to have a script at all. Keep following MDEV-19917. Even a removal cnf and shared library while the server is running shouldn't be a problem. The existence will disappear on restart.

If `UNINSTALL ` is needed, need to check:
* does it fail if tables of this type are open - would cause script to fail. `FLUSH .. WITH READ LOCK` might even prevent the remove.
* need to ensure this is not to be replicated to binary log 

storage/spider/scripts/install_spider.sql is definitely bloated. '.so' vs '.ddl' isn't needed in the syntax. And `install plugin if not exists` is there since 10.4.

https://mariadb.com/kb/en/install-plugin/

so debian/mariadb-plugin-spider.postinst:

mysql ... - e "install plugin if not exists  spider soname 'ha_spider'"

Reply:
MDEV-19917 is now solved with help from @an3l and Spider is now activated through it's own spider.cnf, no install.sql anymore.